File: java-programs/loops/prime-num.java
import java.io.*;
import java.util.Scanner;

class prime_number
{      
    public static void main(String args[])
    {
     int i,n;
     boolean flag = true;
     n = 29;

     for(i=2;i<=n/2;i++)
     {
      if(n%i == 0)
      {
        flag = false;
        break;
      }
     }
     if(flag == true) 
      System.out.println("The Number "+n+" is a prime number.");
    else
      System.out.println("The Number "+n+" is not a prime number.");
    }
}

File: cpp-programs/loops/prime-num.cpp
#include<bits/stdc++.h>
using namespace std;

int main()
{
int n,i,flag=0;

cout<<"Enter the number :::";
cin>>n;

for(i=2;i<n/2;i++)
{
 if(n%i==0)
 {
  flag = 1;
  break;
 }
}

if(flag == 1)
cout<<n<<" is not prime number."<<endl;
else
cout<<n<<" is prime number."<<endl;
}
